reference/programming-model.md

```diff
@@ -189,9 +189,102 @@ url := virtualmachine.DnsName().Apply(func(dnsName string) (interface{}, error)
 
 The `apply` method accepts a callback which will be passed the value of the `Output` when it is available, and which returns the new value.  The result of the call to `apply` is a new `Output` whose value is the value returned from the callback, and which includes the dependencies of the original `Output`.  If the callback itself returns an `Output`, the dependencies of that output are unioned into the dependencies of the returned `Output`.
 
-> Note: The `Output` itself cannot be used directly in string concatenation or other operations, as it is not itself the value of the output.  To transform the value of the output (when it becomes available), the `apply` method should be used instead.
+> Note: The `Output` itself cannot be used directly in string concatenation or other operations, as it is not itself the value of the output.  To transform the value of the output (when it becomes available), the `apply` method should be used instead.  
 
-#### All {#all}
+`.apply` is the most general way to transform one `Output` into another.  However, several common types of transoformations can be done more conveinently.  See (Accessing properties of an Output)[#lifting] and (Working with Outputs and strings)[#ouputs-and-strings] for examples of this.
+
+##### Accessing properties of an Output {#lifting}
+
+It is common to need to only access some subproperty of an `Output` in order to pass in that subproperty to another `Resource`.  For example, when using ACM certificates one might write:
```

Comment:
Perhaps just “property” instead of “subproperty”.

Reply:
done.